### PR TITLE
Chaos gods

### PR DESCRIPTION
**Pull request by OH296**
Added all dialogue and pathways for a basic daemon emissary
-basically just adds place holders for the chaos gods but should give players something to get excited about
Added basic data structs for chaos gods
basic feeders for allegiance to a chaos god 
"quests" for chaos WIP
is stable and ready to merge

errors spotted by @GalacticChimp solved

### CODE REVIEW

**Comment by OH296** [comment]
has no effect on main gameplay, causes no errors

File: objects/obj_controller/Create_0.gml [hunk 1]
```diff
@@ -19,6 +19,28 @@ chaos_rating=0;
 obj_cuicons.alarm[1]=1;// Clean up custom icons
 
 
+// function set to build chaos gods
+chaos_gods = {};
+diplomacy_pathway = "";
+option_selections=[];
+ready=false;
+function build_chaos_gods(){
+	var _god_names = ["Khorne", "Slaanesh", "Nurlgle", "tzeentch"]
```
Comment (GalacticChimp) [inline objects/obj_controller/Create_0.gml:28]: Nur~~l~~gle, **T**zeentch

File: objects/obj_controller/Mouse_50.gml [hunk 2]
```diff
@@ -362,11 +362,120 @@ if (menu=16) and (cooldown<=0){
 }
 
 
+if (menu==20) and (diplomacy==10.1){
+		show_debug_message(diplomacy_pathway, diplo_text);
+		show_debug_message( diplo_text, diplo_text);
+		show_debug_message(cooldown);
+		if (diplomacy_pathway == "intro") and (cooldown <= 0){
+			if point_in_rectangle(mouse_x, mouse_y, option_selections[0].lh, option_selections[0].top, option_selections[0].rh, option_selections[0].base){
```
Comment (GalacticChimp) [inline objects/obj_controller/Mouse_50.gml:370]: can you perhaps extract a method out of this, so you won't have to repeat yourself?

something like this (but with better names):

```
function check_and_assign_demon_pathway_in_intro(index, cooldown_arg, diplo_pathway_name, force_goodbye_arg){
    if point_in_rectangle(mouse_x, mouse_y, option_selections[index].lh, option_selections[index].top, option_selections[index].rh, option_selections[index].base){
                         cooldown=cooldown_arg;
			 diplomacy_pathway =diplo_pathway_name;
			 scr_dialogue(diplomacy_pathway);
                         force_goodbye=force_goodbye_arg;
			}
}
```

**Comment by GalacticChimp** [comment]
looks fine in general, except for the typos

File: objects/obj_controller/Mouse_50.gml (hunk 2 above)
Comment (OH296) [reply objects/obj_controller/Mouse_50.gml:370]: tbh i did actually try something like that but whenever i viewed the screen it just bugged out. probably just my lack of understanding of the gamemaker click events. will play around but idk

File: objects/obj_controller/Create_0.gml (hunk 1 above)
Comment (OH296) [reply objects/obj_controller/Create_0.gml]: sorted

**Review by GalacticChimp — APPROVED** [review]

**Comment by ZAB909** [comment]
fix el conflict

**Pull request merged**